modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/function/FutureDatesFunction.java

```diff
@@ -40,13 +42,23 @@ public Boolean apply(Object object1, Object object2) {
 		LocalDate localDate = DateParameterUtil.getLocalDate(
 			object1.toString());
 
-		SimpleDateFormat formatter = new SimpleDateFormat(
-			"yyyy-MM-dd HH:mm:ss.S");
+		if (object2 instanceof Date) {
+			SimpleDateFormat formatter = new SimpleDateFormat(
+				"yyyy-MM-dd HH:mm:ss.S");
+
+			LocalDate currentDate = DateParameterUtil.getLocalDate(
+				formatter.format(object2));
+
+			if (localDate.isBefore(currentDate)) {
+				return false;
+			}
+
+			return true;
+		}
 
-		LocalDate currentDate = DateParameterUtil.getLocalDate(
-			formatter.format(object2));
+		if (localDate.isBefore(
```

Comment:
hey @ericthmoritsuka 

we could simplify here:
```
LocalDate localDate = DateParameterUtil.getLocalDate(
	object1.toString());

String dateString = object2.toString();

if (object2 instanceof Date) {
	SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd");

	dateString = formatter.format(object2);
}

if (localDate.isBefore(
		DateParameterUtil.getLocalDate(dateString))) {

	return false;
}

return true;
```

What do you think?

Reply:
I feel silly. I put in my mind that I had to be careful with the names of the variables and I completely forgot that I could do just that. Yes, this is a much better solution!